tests/test_dynamo/test_user.py

```diff
@@ -1,73 +1,253 @@
+    }
+    tables.users_table.put_item(Item=original_user_record)
+
+    user = dynamo.user._reset_credits_if_needed(
+        user=original_user_record,
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table,
+    )
+
+    assert user == {'user_id': 'foo', 'remaining_credits': Decimal(25), 'month_of_last_credits_reset': '2024-02'}
+    assert tables.users_table.scan()['Items'] == [user]
+
+
+def test_reset_credits_no_reset(tables, monkeypatch):
+    monkeypatch.setenv('RESET_CREDITS_MONTHLY', 'no')
+
+    original_user_record = {
+        'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-01'
     }
+    tables.users_table.put_item(Item=original_user_record)
 
-    user2 = dynamo.user.create_user('user2')
-    assert user2 == {
-        'user_id': 'user2',
-        'remaining_credits': 25,
+    user = dynamo.user._reset_credits_if_needed(
+        user=original_user_record,
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table,
+    )
+
+    assert user == {'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-01'}
+    assert tables.users_table.scan()['Items'] == [user]
+
+
+def test_reset_credits_same_month(tables, monkeypatch):
+    monkeypatch.setenv('RESET_CREDITS_MONTHLY', 'yes')
+
+    original_user_record = {
+        'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-02'
     }
+    tables.users_table.put_item(Item=original_user_record)
 
-    response = tables.users_table.scan()
-    assert response['Items'] == [user1, user2]
+    user = dynamo.user._reset_credits_if_needed(
+        user=original_user_record,
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table,
+    )
 
-    with pytest.raises(ValueError):
-        dynamo.user.create_user('user1')
+    assert user == {'user_id': 'foo', 'remaining_credits': Decimal(10), 'month_of_last_credits_reset': '2024-02'}
+    assert tables.users_table.scan()['Items'] == [user]
+
+
+def test_reset_credits_infinite_credits(tables, monkeypatch):
+    monkeypatch.setenv('RESET_CREDITS_MONTHLY', 'yes')
+
+    original_user_record = {
+        'user_id': 'foo', 'remaining_credits': None, 'month_of_last_credits_reset': '2024-01'
+    }
+    tables.users_table.put_item(Item=original_user_record)
+
+    user = dynamo.user._reset_credits_if_needed(
+        user=original_user_record,
+        default_credits=Decimal(25),
+        current_month='2024-02',
+        users_table=tables.users_table,
+    )
+
+    assert user == {'user_id': 'foo', 'remaining_credits': None, 'month_of_last_credits_reset': '2024-01'}
+    assert tables.users_table.scan()['Items'] == [user]
+
+
+def test_reset_credits_failed_same_month(tables, monkeypatch):
```

Comment:
`wrong_month` might be more accurate, since we're testing `2024-02` vs `2024-01`